Add _reload_search_analyzers endpoint to HLRC

cbuescher opened the pull request:
This change adds the new endpoint that allows reloading of search analyzers to
the high-level java rest client.

Relates to #43313

Commit:
Add _reload_search_analyzers endpoint to HLRC

This change adds the new endpoint that allows reloading of search analyzers to
the high-level java rest client.

Relates to #43313

elasticmachine commented:
Pinging @elastic/es-search

elasticmachine commented:
Pinging @elastic/es-core-features

cbuescher commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into hlrc-reload-analyzers

cbuescher commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into hlrc-reload-analyzers

jimczi commented on docs/java-rest/high-level/indices/reload_analyzers.asciidoc:
```diff
@@ -0,0 +1,45 @@
+--------------------------------------------------
+include-tagged::{doc-tests-file}[{api}-response]
+--------------------------------------------------
+<1> Shard statistics including total number of shards the request was executed on
```
This is not entirely related to this pr but I find it confusing that we return a `Shards` object in the response even though the action is executed at the node level. I don't have a good way to fix this but we should at least document here that the statistics are per node/index and not per shards ?

cbuescher replied:
I agree, this is not ideal. I'll look into the options of either documenting it better, change the naming of the object only for the java api level or whether maybe it makes sense change or even leave out the whole "shard: { ... }". That would have to be done further down the calls stack though.

cbuescher replied:
I looked into the options, and it would be possible to remove the "shard: { ... }" element from the REST response completely even though TransportReloadAnalyzersAction would still be able to extend BroadcastResponse, but we could overwrite "toXContent" without outputing broadcast shards header. Then again, potential failure exceptions are currently also written inside the "shards" header, replacing that toXContent rendering and parsing might turn out to be a bigger change than intended.
Changing the accessor or object name would also involve either not being able to extend BroadcastResponse on the HLRC client side (this would mean more custom parsing code, other getters etc... for failures), so I'll push an update to the docs for now. If you want to discuss the other options again let me know.

jimczi reviewed (commented):
code LGTM, I left one comment regarding the `shards` object that is returned in the response.

Commit:
Extended shard statistics doc

Commit:
Merge branch 'master' into hlrc-reload-analyzers

Commit:
Return reloaded analyzers in _reload_search_ananlyzer response

Currently the repsonse of the "_reload_search_analyzer" endpoint contains the
index names and nodeIds of indices were analyzers reloading was triggered. This
change add the names of the search-time analyzers that were reloaded.

Closes #43804

Commit:
Merge branch 'master' into hlrc-reload-analyzers

Commit:
Adding realoading details to HLRC response

Commit:
Adding to docs

cbuescher commented:
@jimczi as this PR depends on #43813 and CI is on strike right now I cherry-picked that PR in 
3578870 (assuming is goes in like this) and added the necessary changes for the additional info returned by the response in the HLRC response class and testing. I think this is ready for another review even though we don't get testing atm.

cbuescher commented:
@elasticmachine test this please

cbuescher commented:
@elasticmachine run elasticsearch-ci/1

Commit:
Merge branch 'master' into hlrc-reload-analyzers

cbuescher commented:
@elasticmachine run elasticsearch-ci/2

cbuescher commented:
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/packaging-sample

cbuescher commented:
@elasticmachine run elasticsearch-ci/2
@elasticmachine run elasticsearch-ci/packaging-sample

jimczi reviewed (approved):
LGTM
